concordium-contracts-common/src/schema.rs

```diff
@@ -1545,14 +1575,58 @@ mod tests {
     fn test_module_v2_serial_deserial_is_id() {
         let m = ModuleV2 {
             contracts: BTreeMap::from([("a".into(), ContractV2 {
-                init:    Some(FunctionV2::Error(Type::U8)),
+                init:    Some(FunctionV2 {
+                    parameter:    Some(Type::String(SizeLength::U32)),
+                    return_value: Some(Type::String(SizeLength::U32)),
+                    error:        Some(Type::String(SizeLength::U32)),
+                }),
                 receive: BTreeMap::from([
-                    ("b".into(), FunctionV2::Rv(Type::String(SizeLength::U32))),
-                    ("c".into(), FunctionV2::ParamError {
-                        parameter: Type::U8,
-                        error:     Type::Bool,
+                    ("b".into(), FunctionV2 {
+                        parameter:    Some(Type::String(SizeLength::U32)),
+                        return_value: Some(Type::String(SizeLength::U32)),
+                        error:        Some(Type::String(SizeLength::U32)),
+                    }),
+                    ("c".into(), FunctionV2 {
+                        parameter:    Some(Type::String(SizeLength::U32)),
+                        return_value: Some(Type::String(SizeLength::U32)),
+                        error:        Some(Type::String(SizeLength::U32)),
+                    }),
+                ]),
+            })]),
+        };
+
+        assert_eq!(serial_deserial(&m), Ok(m));
+    }
+
+    #[test]
+    fn test_module_v3_serial_deserial_is_id() {
+        let mut event_map = BTreeMap::new();
+        let tag: u8 = 1;
+        event_map.insert(
+            tag,
+            (String::from("EventOne"), Fields::Named(vec![(String::from("value"), Type::U8)])),
+        );
+
+        let m = ModuleV3 {
+            contracts: BTreeMap::from([("a".into(), ContractV3 {
+                init:    Some(FunctionV2 {
+                    parameter:    Some(Type::String(SizeLength::U32)),
+                    return_value: Some(Type::String(SizeLength::U32)),
+                    error:        Some(Type::String(SizeLength::U32)),
+                }),
+                receive: BTreeMap::from([
+                    ("b".into(), FunctionV2 {
+                        parameter:    Some(Type::String(SizeLength::U32)),
+                        return_value: Some(Type::String(SizeLength::U32)),
+                        error:        Some(Type::String(SizeLength::U32)),
+                    }),
+                    ("c".into(), FunctionV2 {
+                        parameter:    Some(Type::String(SizeLength::U32)),
+                        return_value: Some(Type::String(SizeLength::U32)),
+                        error:        Some(Type::String(SizeLength::U32)),
                     }),
                 ]),
+                event:   Some(Type::TaggedEnum(event_map)),
             })]),
         };
```

Comment:
There does not appear to be any test here. I suggest adding a general randomized test

```rust
    #[test]
    fn test_module_v3_schema_serial_deserial_is_id() {
        use rand::prelude::*;
        use rand_pcg::Pcg64;

        let seed: u64 = random();
        let mut rng = Pcg64::seed_from_u64(seed);
        let mut data = [0u8; 100000];
        rng.fill_bytes(&mut data);

        let mut unstructured = Unstructured::new(&data);

        for _ in 0..10000 {
            let schema = ModuleV3::arbitrary(&mut unstructured).unwrap();

            let res = from_bytes::<ModuleV3>(&to_bytes(&schema)).unwrap();
            assert_eq!(schema, res);
        }
    }
```

You will need to add 
```
#[cfg_attr(test, derive(arbitrary::Arbitrary))]
```
to ModuleV3, ContractV3